### PR TITLE
flatpak: adopt; modernize

### PR DESCRIPTION
**Pull request by getchoo**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by getchoo** [comment]
the pkgs/by-name check seems to be a false positive. `flatpak.tests.validate-icon` runs as expected

File: pkgs/by-name/fl/flatpak/package.nix [hunk 1]
```diff
@@ -175,37 +188,42 @@ stdenv.mkDerivation (finalAttrs: {
...
       inherit (builtins) storeDir;
     };
 
+    updateScript = nix-update-script { };
+
     tests = {
       installedTests = nixosTests.installed-tests.flatpak;
 
       validate-icon = runCommand "test-icon-validation" { } ''
         ${finalAttrs.finalPackage}/libexec/flatpak-validate-icon --sandbox 512 512 ${../../../applications/audio/zynaddsubfx/ZynLogo.svg} > "$out"
```
Comment (jtojnar) [inline pkgs/by-name/fl/flatpak/package.nix:213]: Maybe we can try switching to `"${nixos-icons}/share/icons/hicolor/scalable/apps/nix-snowflake-white.svg"` to fix the check.

File: pkgs/by-name/fl/flatpak/package.nix [hunk 2]
```diff
@@ -1,75 +1,90 @@
...
+    "man"
+    "doc"
+    "devdoc"
+    "installedTests"
+  ];
 
   src = fetchurl {
     url = "https://github.com/flatpak/flatpak/releases/download/${finalAttrs.version}/flatpak-${finalAttrs.version}.tar.xz";
-    sha256 = "sha256-U482ssb4xw7v0S0TrVsa2DCCAQaovTqfa45NnegeSUY="; # Taken from https://github.com/flatpak/flatpak/releases/
+    hash = "sha256-U482ssb4xw7v0S0TrVsa2DCCAQaovTqfa45NnegeSUY="; # Taken from https://github.com/flatpak/flatpak/releases/
```
Comment (jtojnar) [inline pkgs/by-name/fl/flatpak/package.nix:74]: This comment no longer applies.

**Review by jtojnar — COMMENTED** [review]
Thanks for adopting this.

Just FYI, timezone handling is currently broken (see e.g. https://gitlab.gnome.org/GNOME/gnome-calendar/-/issues/1048). There is an [upstream patch](https://github.com/flatpak/flatpak/pull/5850) that might fix it but it is currently untested.

**Comment by getchoo** [comment]
> Just FYI, timezone handling is currently broken (see e.g. https://gitlab.gnome.org/GNOME/gnome-calendar/-/issues/1048). There is an https://github.com/flatpak/flatpak/pull/5850 that might fix it but it is currently untested.

thanks for the heads up, i hadn't heard about this before. i'll try testing the patch out myself and add it in a follow up PR

**Comment by jtojnar** [comment]
Merged the other PR but now there is a conflict.

**Comment by jtojnar** [comment]
Thanks again.

**Pull request merged**